### PR TITLE
FALCON-1892: Remove client side Recipe logic

### PR DESCRIPTION
**Pull request by sowmyaramesh**

### CODE REVIEW

File: addons/extensions/hive-mirroring/src/main/resources/runtime/hive-mirroring-secure-workflow.xml [hunk 1]
```diff
@@ -142,10 +142,8 @@
             <arg>${clusterForJobRunWriteEP}</arg>
             <arg>-clusterForJobNNKerberosPrincipal</arg>
             <arg>${clusterForJobNNKerberosPrincipal}</arg>
-            <arg>-tdeEncryptionEnabled</arg>
```
Comment (peeyushb) [inline addons/extensions/hive-mirroring/src/main/resources/runtime/hive-mirroring-secure-workflow.xml]: TDE encryption attributes removed from secure mirroring workflow. Is this expected.

File: addons/extensions/hive-mirroring/src/main/resources/runtime/hive-mirroring-secure-workflow.xml [hunk 2]
```diff
@@ -242,14 +240,10 @@
...
-            <arg>${tdeEncryptionEnabled}</arg>
```
Comment (peeyushb) [inline addons/extensions/hive-mirroring/src/main/resources/runtime/hive-mirroring-secure-workflow.xml]: same comment as above.

File: addons/extensions/hive-mirroring/src/main/resources/runtime/hive-mirroring-secure-workflow.xml [hunk 3]
```diff
@@ -242,14 +240,10 @@
...
-            <arg>-drJobName</arg>
-            <arg>${drJobName}-${nominalTime}</arg>
+            <arg>-jobName</arg>
+            <arg>${jobName}-${nominalTime}</arg>
             <arg>-executionStage</arg>
             <arg>export</arg>
-            <arg>-counterLogDir</arg>
-            <arg>${logDir}/job-${nominalTime}/${srcClusterName == 'NA' ? '' : srcClusterName}/</arg>
```
Comment (peeyushb) [inline addons/extensions/hive-mirroring/src/main/resources/runtime/hive-mirroring-secure-workflow.xml]: counterLogDir attribute removed from mirroring workflow xml. Is this expected.

File: addons/extensions/hive-mirroring/src/main/resources/runtime/hive-mirroring-workflow.xml [hunk 4]
```diff
@@ -154,12 +154,10 @@
             <arg>${clusterForJobRun}</arg>
             <arg>-clusterForJobRunWriteEP</arg>
             <arg>${clusterForJobRunWriteEP}</arg>
-            <arg>-drJobName</arg>
-            <arg>${drJobName}-${nominalTime}</arg>
+            <arg>-jobName</arg>
+            <arg>${jobName}-${nominalTime}</arg>
             <arg>-executionStage</arg>
             <arg>export</arg>
-            <arg>-counterLogDir</arg>
```
Comment (peeyushb) [inline addons/extensions/hive-mirroring/src/main/resources/runtime/hive-mirroring-workflow.xml]: counterLogDir attribute removed from mirroring workflow xml. Is this expected.

File: extensions/src/main/java/org/apache/falcon/extensions/util/ExtensionProcessBuilderUtils.java [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+        }
+
+        String retryDelay = extensionProperties.getProperty(ExtensionProperties.RETRY_DELAY.getName());
+        if (StringUtils.isNotBlank(retryDelay)) {
+            processRetry.setDelay(Frequency.fromString(retryDelay));
+        } else {
+            processRetry.setDelay(defaultRetryDelay);
+        }
+
+//        String retryOnTimeout = extensionProperties.getProperty(ExtensionProperties.RETRY_ON_TIMEOUT.getName());
```
Comment (peeyushb) [inline extensions/src/main/java/org/apache/falcon/extensions/util/ExtensionProcessBuilderUtils.java]: remove the commented code.

File: oozie/src/main/java/org/apache/falcon/service/SharedLibraryHostingService.java [hunk 6]
```diff
@@ -75,16 +80,77 @@ public String getJarName(Path path) {
         }
     };
 
-    private void addLibsTo(Cluster cluster) throws FalconException {
-        FileSystem fs = null;
+    private void pushExtensionArtifactsToCluster(final Cluster cluster,
```
Comment (peeyushb) [inline oozie/src/main/java/org/apache/falcon/service/SharedLibraryHostingService.java]: Some of the comments in the function is not required.

File: addons/extensions/hdfs-mirroring/README [hunk 7]
```diff
@@ -14,10 +14,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-HDFS Directory Replication Recipe
+HDFS Directory Replication Extension
 
 Overview
-This recipe implements replicating arbitrary directories on HDFS from one
+This extesion implements replicating arbitrary directories on HDFS from one
```
Comment (peeyushb) [inline addons/extensions/hdfs-mirroring/README]: minor nit: "extension"

**Comment by PraveenAdlakha** [comment]
@sowmyaramesh  please update the existing documentation..

**Comment by sowmyaramesh** [comment]
I wanted to create two pull requestions on for removing client side and other for moving it to server side. Somehow everything in one pull request. I will fix that tomorrow. Sorry about that. Please continue review after In update the pull requests.

**Comment by sowmyaramesh** [comment]
@PraveenAdlakha : I should have mentioned in the jira. Planning to update documentation as part of https://issues.apache.org/jira/browse/FALCON-1106. Please ignore documentation errors in this pull request if any.

**Comment by sowmyaramesh** [comment]
Closing this as I have created two separate pull requests https://github.com/apache/falcon/pull/91 and https://github.com/apache/falcon/pull/92.

**Pull request closed without merging**